Initial version of Dockerfile for Boolector

andrewvaughanj opened the pull request:
Fix for Issue #33.

Tested as working, and passing the tests, with the latest CaDiCaL.

mpreiner commented on contrib/docker/DOCKER.md:
````diff
@@ -0,0 +1,99 @@
+```
+
+*Note*: the current working directory (`pwd`) has been mapped to `/tmp` within
+the container -- hence why the file-path to the SMT2LIB file you wish to run
````
```suggestion
the container -- hence why the file-path to the SMT-LIB v2 file you wish to run
```

andrewvaughanj replied:
Done!

mpreiner commented on contrib/docker/DOCKER.md:
```diff
@@ -0,0 +1,99 @@
+files via the use of a Docker container. The base container used is Alpine
+Linux.
+
+This guide presumes you are familiar with both Docker and Boolector, and are
```
I'm not a native speaker, but here is another suggestion (not sure if presume is the right choice here?)

```suggestion
This guide assumes the reader is familiar with both Docker and Boolector, and is
```

andrewvaughanj replied:
Done!

mpreiner commented on contrib/docker/DOCKER.md:
````diff
@@ -0,0 +1,99 @@
+```
+#
+# Presuming you're at the root of a Boolector clone
+#
````
Maybe remove the empty comment lines (to make it more compact)?

andrewvaughanj replied:
Done!

mpreiner commented on contrib/docker/DOCKER.md:
````diff
@@ -0,0 +1,99 @@
+```
+#
+# Presuming you're at the root of a Boolector clone
+#
````
Ditto.

andrewvaughanj replied:
Done!

mpreiner commented on contrib/docker/DOCKER.md:
````diff
@@ -0,0 +1,99 @@
+```
+#
+# Presuming you're at the root of a Boolector clone
+#
````
Ditto.

andrewvaughanj replied:
Done!

mpreiner commented on contrib/docker/DOCKER.md:
```diff
@@ -0,0 +1,99 @@
+
+#
+# Run a Python script using pyboolector
+#
```
Ditto.

andrewvaughanj replied:
Done!

mpreiner commented on contrib/docker/DOCKER.md:
```diff
@@ -0,0 +1,99 @@
+
+#
+# You are now inside of the container!!!
+#
```
Ditto. Maybe merge with below comment?

andrewvaughanj replied:
Done!

mpreiner commented on contrib/docker/DOCKER.md:
```diff
@@ -0,0 +1,99 @@
+
+#
+# Leave the container!!!
+#
```
Ditto.

andrewvaughanj replied:
Done!

andrewvaughanj commented:
@mpreiner thanks for the comments -- all done.

There are two changes since you last reviewed this:

1. The Dockerfile used to `git clone` Boolector into itself -- this wasn't ideal, as it meant you couldn't build your own local branch inside of Docker, only the current master. This is now changed the Dockerfile copies the current repository into itself before building (which means you can build your own local branches). The documentation has slightly changed to reflect this modified build step.

2. I rebased this branch on top of my the latest Windows patches branches -- without these updates, you couldn't build CaDiCaL inside of Docker. I _have not_ squashed this commit. If the Windows patches go into master first, I can then rebase this onto of master and /voila/.

mpreiner commented on contrib/docker/Dockerfile:
```diff
@@ -0,0 +1,98 @@
+
+#
+# Copy our current source tree to the Docker image
+#
```
I know you changed this in order to be able to support building local branches. I would still prefer to define `ARG RELEASE=master` and `wget -q https://github.com/boolector/boolector/archive/${RELEASE}.tar.gz -O ...` from the official repository since this will be the more common use case. You can add this command as a comment and add a comment like `If you want to build a local branch of Boolector add these lines and comment out the others...`.

andrewvaughanj replied:
Okay, this is done.

I removed the ability to build the local source tree (making this conditional was more effort than it was worth).

You can now provide Docker build arguments of `UPSTREAM` and `BRANCH` to say where to build from (default is `https://github.com/Boolector/boolector/archive` and `master`).

I add this to the documentation to provide an example of how to build from the non-default upstream/branch.

aniemetz commented:
@andrewvaughanj @mpreiner what is the status of this one?

andrewvaughanj commented:
Not ready -- doesn't work with latest master.
